raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -173,25 +178,40 @@ private void updateFollowerHighWatermark(FollowerState state, OptionalLong highW
             long newHighWatermark = Math.min(endOffset().offset, highWatermark);
             state.updateHighWatermark(OptionalLong.of(newHighWatermark));
             highWatermarkOpt.ifPresent(log::updateHighWatermark);
-            logger.trace("Follower high watermark updated to {}", newHighWatermark);
+            logger.debug("Follower high watermark updated to {}", newHighWatermark);
             applyCommittedRecordsToStateMachine();
         });
     }
 
-    private void updateLeaderEndOffset(LeaderState state) {
+    private void updateLeaderEndOffsetAndTimestamp(LeaderState state) {
         if (state.updateLocalEndOffset(log.endOffset())) {
-            logger.trace("Leader high watermark updated to {} after end offset updated to {}",
+            logger.debug("Leader high watermark updated to {} after end offset updated to {}",
                     state.highWatermark(), log.endOffset());
             applyCommittedRecordsToStateMachine();
         }
+
+        maybeUpdateFetchTimerWithRemoteFetchTimestamp(state, state.localId(), time.milliseconds());
+    }
+
+    private void maybeUpdateFetchTimerWithRemoteFetchTimestamp(LeaderState state, int replicaId, long timestamp) {
+        final OptionalLong lastFetchTimestamp = state.updateFetchTimestamp(replicaId, timestamp);
```

Comment:
nit: for consistency with `updateReplicaEndOffset`, maybe we can just inline this method in `updateLeaderEndOffsetAndTimestamp`. Also, maybe we can use `updateReplicaEndOffsetAndTimestamp`?

Reply:
Ack.

Reply:
I ended up letting both `updateReplicaXX` and `updateLeaderXX` call this function instead of inline it.